drivers/i2c/i2c_ll_stm32.c

```diff
@@ -300,6 +305,10 @@ static const struct i2c_stm32_config i2c_stm32_cfg_##name = {		\
 	},								\
 	STM32_I2C_IRQ_HANDLER_FUNCTION(name)				\
 	.bitrate = DT_PROP(DT_NODELABEL(name), clock_frequency),	\
+	COND_CODE_1(CONFIG_I2C_STM32_V2,				\
+	(.timings_list = (const struct i2c_config_timing *) i2c_timings_##name,\
```

Comment:
why is cast needed here?

Reply:
i2c_timings_##name is a table of uint32_t  retrieved from the device tree array.
Even though the device tree array is of form <a b>, <c d> it results in a table of {a, b, c, d}, not {{a, b}, {c, d}} 
therefore I cannot make i2c_timings_##name a table of const struct i2c_config_timing - I tried and I would get a warning. so I decided to get this uint32_t table and cast it to the our structured format.

Reply:
@gmarull does that answer your question  ? would you have a alternative proposal ?

Reply:
Got it, thanks for the clarification